contexts/WalletManagerContext.tsx

```diff
@@ -660,10 +668,52 @@ export function WalletManagerProvider({ children }: { children: ReactNode }) {
 
     fetchBalance();
 
-    rpcClient.addEventListener("utxos-changed", fetchBalance);
+    function checkIncomingUtxos(event: {
+      added: UtxoEntryReference[];
+      removed: UtxoEntryReference[];
+    }) {
+      if (event.added.length === 0) {
+        return;
+      }
+
+      const txId = event.added[0].outpoint.transactionId;
+      const network = settings?.networkId ?? NetworkType.Mainnet;
+      const explorerTxLink = explorerTxLinks[network];
+      const outgoingAmount = event.removed.reduce(
+        (acc, curr) =>
+          addresses.includes(curr.address?.toString() ?? "")
+            ? acc + curr.amount
+            : acc,
+        0n,
+      );
+      const incomingAmount = event.added.reduce(
+        (acc, curr) =>
+          addresses.includes(curr.address?.toString() ?? "")
+            ? acc + curr.amount
+            : acc,
+        0n,
+      );
+      const transferAmount = incomingAmount - outgoingAmount;
+      if (transferAmount <= 0) {
+        return;
+      }
+
+      internalToast.info(
+        `You’ve received ${sompiToKaspaString(transferAmount)} amount of KAS. Click to open on explorer`,
+        () => browser.tabs.create({ url: `${explorerTxLink}${txId}` }),
+        txId,
+      );
+    }
+
+    rpcClient.addEventListener("utxos-changed", async (event) => {
+      await fetchBalance();
+      checkIncomingUtxos(event.data);
+    });
+
     rpcClient.subscribeUtxosChanged(addresses);
     return () => {
       rpcClient.unsubscribeUtxosChanged(addresses);
+
       rpcClient.removeEventListener("utxos-changed", fetchBalance);
```

Comment:
The `removeEventListener` is not updated.

Reply:
Please check this commit:
https://github.com/forbole/kastle/pull/33/commits/a7884563e31c85ce662bdec078280475e646d30e

Reply:
Thanks for catching that bug !!!

Also, I have removed the `safeQueue` because it created more renders than without. I think it was solving the issues of not accounting for the settings loading.
Let me know if I am missing anything. I tested a bit it seems to work fine.

797ad2242009605f690506d73d913410c6ed68fa

Reply:
Yeah, I agree with it. Javascript event loop have already queued the operations. No need to have this.